### PR TITLE
ESC-341 subsidy payment page

### PR DESCRIPTION
**Pull request by priyankadhiman3006**

### CODE REVIEW

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/views/AddClaimAmountPage.scala.html [hunk 1]
```diff
@@ -55,21 +56,25 @@
   @formHelper(action = controllers.routes.SubsidyController.postAddClaimAmount()) {
 
     @H1(messages("add-claim-amount.title"))
-    @P(messages("add-claim-amount.p1"))
-    @P(messages("add-claim-amount.p2"))
+    @P(messages("add-claim-amount.p1", "https://ec.europa.eu/info/funding-tenders/procedures-guidelines-tenders/information-contractors-and-beneficiaries/exchange-rate-inforeuro_en"))
```
Comment (sam-miele) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/views/AddClaimAmountPage.scala.html:59]: Should we consider externalising this url?

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/views/AddClaimAmountPage.scala.html [hunk 2]
```diff
@@ -55,21 +56,25 @@
...
+      </ul>
 
     @govukInput(Input(
       id = "claim-amount",
       name = "claim-amount",
       prefix = Some(PrefixOrSuffix(content = Text("€"))),
       value = form.data.get("claim-amount"),
       errorMessage = if (form.hasErrors) {
-        Some(ErrorMessage(content = Text(messages("add-claim-amount." + form.errors.head.message))))
+          Some(ErrorMessage(content = Text(messages(s"$key." + form.errors.head.message))))
```
Comment (sam-miele) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/views/AddClaimAmountPage.scala.html:72]: NABD - `messages(s"$key.${form.errors.head.message}")`

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/views/AddClaimAmountPage.scala.html (hunk 1 above)
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/views/AddClaimAmountPage.scala.html:59]: it was like this so I got lazy and copy pasted  🤣 . I can do that

Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/views/AddClaimAmountPage.scala.html]: updated [here](https://github.com/hmrc/eu-subsidy-compliance-frontend/pull/78/commits/7c6d711ea23139e0d75dac92ebcb27d02364c37d)

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/views/AddClaimAmountPage.scala.html (hunk 2 above)
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/views/AddClaimAmountPage.scala.html]: [updated](https://github.com/hmrc/eu-subsidy-compliance-frontend/pull/78/commits/7c6d711ea23139e0d75dac92ebcb27d02364c37d)

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/config/AppConfig.scala [hunk 3]
```diff
@@ -26,8 +26,9 @@ class AppConfig @Inject()(config: Configuration) {
   val ggSignInUrl:        String = config.get[String](s"urls.ggSignInUrl")
   val ggSignOutUrl:       String = config.get[String](s"urls.ggSignOutUrl")
   val eccEscSubscribeUrl: String = config.get[String](s"urls.eccEscSubscribeUrl")
+  val exchangeRateToolUrl: String = config.get[String](s"urls.exchangeRateToolUrl")
 
-  val betaFeedbackUrlNoAuth: String = "TODO" // TODO
+  val betaFeedbackUrlNoAuth: String = "TODO" // TODO§
```
Comment (carwynellis) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/config/AppConfig.scala:31]: Minor - typo?

File: conf/application.conf [hunk 4]
```diff
@@ -165,6 +165,7 @@ urls {
   ggSignOutUrl = "http://localhost:9553/bas-gateway/sign-out-without-state?continue=http://localhost:8994"
   eccEscSubscribeUrl = "http://localhost:6750/customs-enrolment-services/esc/subscribe"
   userResearchUrl = "#"
+  exchangeRateToolUrl = "https://ec.europa.eu/info/funding-tenders/procedures-guidelines-tenders/information-contractors-and-beneficiaries/exchange-rate-inforeuro_en"
```
Comment (carwynellis) [inline conf/application.conf:168]: Should be `exchangeRateToUrl`

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/SubsidyController.scala [hunk 5]
```diff
@@ -104,27 +104,37 @@ class SubsidyController @Inject()(
...
   def getClaimAmount: Action[AnyContent] = escAuthentication.async { implicit request =>
     implicit val eori: EORI = request.eoriNumber
-    // TODO - add 'getPrevious to all'
-    store.get[SubsidyJourney].flatMap {
-      case Some(journey) => journeyTraverseService.getPrevious[SubsidyJourney].map { p =>
-        val form = journey.claimAmount.value.fold(claimAmountForm)(claimAmountForm.fill)
-        Ok(addClaimAmountPage(form, p))
-      }
-      case _ => handleMissingSessionData("Subsidy journey")
+    val result: OptionT[Future, Result] = for {
```
Comment (carwynellis) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/SubsidyController.scala:107]: 👍

**Review by carwynellis — APPROVED** [review]
Looks fine but please fix the typo in the config.

File: conf/application.conf (hunk 4 above)
Comment (sam-miele) [reply conf/application.conf:168]: I read this as "The url to the exchange rate tool?" - which would make `exchangeRateToolUrl` correct. Priyanka can let us know.

Comment (priyankadhiman3006) [reply conf/application.conf:168]: Hi Sam, yes you are correct . @carwynellis it's exchange rate tool we are redirecting to, so I think exchangeRateToolUrl is ok

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/config/AppConfig.scala (hunk 3 above)
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/config/AppConfig.scala]: fixed

File: conf/application.conf (hunk 4 above)
Comment (carwynellis) [reply conf/application.conf:168]: Haha! I need more coffee!

**Pull request merged**